api/src/test/java/uk/gov/dhsc/htbhf/claimant/entitlement/AdditionalPregnancyVoucherCalculatorTest.java

```diff
@@ -0,0 +1,70 @@
+package uk.gov.dhsc.htbhf.claimant.entitlement;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.dhsc.htbhf.claimant.entity.PaymentCycle;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.PaymentCycleTestDataFactory.aPaymentCycleWithStartAndEndDate;
+
+@ExtendWith(MockitoExtension.class)
+class AdditionalPregnancyVoucherCalculatorTest {
+
+    private static final int NUMBER_OF_CALCULATION_PERIODS = 4;
+    private static final int ENTITLEMENT_CALCULATION_DURATION = 7;
+    private static final Integer VOUCHERS_PER_PREGNANCY = 1;
+
+    @Mock
+    private PregnancyEntitlementCalculator pregnancyEntitlementCalculator;
+
+    @Mock
+    private PaymentCycleConfig paymentCycleConfig;
+
+    private AdditionalPregnancyVoucherCalculator calculator;
+
+    @BeforeEach
+    void setUp() {
+        given(paymentCycleConfig.getEntitlementCalculationDurationInDays()).willReturn(ENTITLEMENT_CALCULATION_DURATION);
+        calculator = new AdditionalPregnancyVoucherCalculator(VOUCHERS_PER_PREGNANCY, paymentCycleConfig, pregnancyEntitlementCalculator);
+    }
+
+    @ParameterizedTest
+    @CsvSource({
+            "-5, 0", // no vouchers if the claim updated date is before the payment cycle start date
+            "0, 0",  // if on the same day as the payment cycle then no need for additional vouchers as the payment cycle will now include pregnancy vouchers.
+            "1, 3",  // week one, get vouchers for week two, three and four
+            "8, 2",  // week two, get vouchers for week three and four
+            "15, 1", // week three, get vouchers for week four
+            "22, 0"  // week four, get no vouchers
+    })
+    void shouldCalculateNumberOfAdditionalPregnancyVouchers(int daysInCycle, int expectedNumberOfVouchers) {
```

Comment:
The variable `daysInCycle` doesn't make sense, especially as the first value is -5, you can't have a negative cycle length.

Reply:
It's the day after the cycle was created, the comments should explain but I'll rename the variable as well.